app/controllers/api/v2/preupgrade_reports_controller.rb

```diff
@@ -3,17 +3,17 @@
 module Api
   module V2
     class PreupgradeReportsController < ::Api::V2::BaseController
-      before_action :find_resource, only: %i[show]
-
       api :GET, '/preupgrade_reports/', N_('List Preupgrade reports')
       param_group :search_and_pagination, ::Api::V2::BaseController
       def index
         @preupgrade_reports = resource_scope_for_index
       end
 
-      api :GET, '/preupgrade_reports/:id', N_('Show Preupgrade report')
+      api :GET, '/job_invocations/:id/preupgrade_report', N_('Show Preupgrade report')
       param :id, :identifier, required: true
-      def show; end
+      def show
+        @preupgrade_report = PreupgradeReport.find_by(job_invocation_id: params[:id])
```

Comment:
Shouldn't this check if the user has permissions to see the preupgrade report?

Reply:
Permission is already checked on controller level.

Reply:
In REX we still check if the user has permissions to view a specific host when looking at job details (see https://github.com/theforeman/foreman_remote_execution/blob/master/app/controllers/job_invocations_controller.rb#L56). Would it make sense to limit what preupgrade report entries the user can see based on `view_hosts` permission?

@ares opinions?

Reply:
yeah that makes sense, this is the case where we need to check that user has permission for both view_job_invocation and view_hosts I think